lib/wlib/strings/DynamicString.cpp

```diff
@@ -0,0 +1,482 @@
+    *
+    * @param lhs @code DynamicString @endcode string as left hand side string
+    * @param rhs character string as right hand side string
+    * @return true or false based on if two strings are equal
+    */
+    bool operator==(const DynamicString &lhs, const char *rhs) {
+        return lhs.compare(rhs) == 0;
+    }
+
+    /**
+    * Comparative operator compares if string and single character are equal or not
+    *
+    * @param lhs character as left hand side
+    * @param rhs @code DynamicString string as right hand side string
+    * @return true or false based on if two strings are equal
+    */
+    bool operator==(const char lhs, const DynamicString &rhs) {
+        return rhs.compare(lhs) == 0;
+    }
+
+    /**
+    * Comparative operator compares if string and single character are equal or not
+    *
+    * @param lhs @code DynamicString string as left hand side string
+    * @param rhs character as right hand side
+    * @return true or false based on if two strings are equal
+    */
+    bool operator==(const DynamicString &lhs, const char rhs) {
+        return lhs.compare(rhs) == 0;
+    }
+
+    /**
+    * Additive operator adds two given strings
+    *
+    * @param lhs @code DynamicString string as left hand side string
+    * @param rhs @code DynamicString string as right hand side string
+    * @return a @code DynamicString that is build from left hand string and right hand string
+    */
+    DynamicString operator+(const DynamicString &lhs, const DynamicString &rhs) {
+        DynamicString newStr(lhs.c_str());
+        newStr.append(rhs);
+        return newStr;
+    }
+
+    /**
+    * Additive operator adds two given strings
+    *
+    * @param lhs character string as left hand side string
+    * @param rhs @code DynamicString string as right hand side string
+    * @return a @code DynamicString that is build from left hand string and right hand string
+    */
+    DynamicString operator+(const char *lhs, const DynamicString &rhs) {
+        DynamicString newStr(lhs);
+        newStr.append(rhs);
+        return newStr;
+    }
+
+    /**
+    * Additive operator adds two given strings
+    *
+    * @param lhs @code DynamicString string as left hand side string
+    * @param rhs character string as right hand side string
+    * @return a @code DynamicString that is build from left hand string and right hand string
+    */
+    DynamicString operator+(const DynamicString &lhs, const char *rhs) {
+        DynamicString newStr(lhs.c_str());
+        newStr.append(rhs);
+        return newStr;
+    }
+
+    /**
+    * Additive operator single character to string
+    *
+    * @param lhs character as left hand side
+    * @param rhs @code DynamicString string as right hand side string
+    * @return a @code DynamicString that is build from left hand string and right hand string
+    */
+    DynamicString operator+(const char lhs, const DynamicString &rhs) {
+        char temp[1] = {lhs};

```

Comment:
actually looks like this is what caused the test to fail, should also have `char temp[2] = {lhs, '\0'}`